pandas/core/series.py

```diff
@@ -5256,6 +5257,29 @@ def to_period(self, freq=None, copy=True) -> Series:
             self, method="to_period"
         )
 
+    @deprecate_nonkeyword_arguments(version=None, allowed_args=["self", "method"])
+    def interpolate(
+        self: Series,
+        method: str = "linear",
+        axis: Axis = 0,
+        limit: int | None = None,
+        inplace: bool = False,
+        limit_direction: str | None = None,
+        limit_area: str | None = None,
+        downcast: str | None = None,
+        **kwargs,
+    ) -> Series | None:
+        return super().interpolate(
+            method,
+            axis,
+            limit,
+            inplace,
+            limit_direction,
+            limit_area,
+            downcast,
+            **kwargs,
+        )
+
```

Comment:
if deprecate_nonkeyword_arguments is updated to accept a class argument then maybe could do something like

```
    interpolate = deprecate_nonkeyword_arguments(
        version=None, allowed_args=["self", "method"], class="DataFrame"
    )(NDFrame.interpolate)
```